Marlin/stepper.cpp

```diff
@@ -91,7 +91,7 @@ volatile uint32_t Stepper::step_events_completed = 0; // The number of step even
 
 #if ENABLED(ADVANCE) || ENABLED(LIN_ADVANCE)
 
-  unsigned char Stepper::old_OCR0A;
+  unsigned char Stepper::old_OCR0A = 0;
```

Comment:
It differs for automatic, stack-based variables, which are not initialized to zero the way that globals and statics are.